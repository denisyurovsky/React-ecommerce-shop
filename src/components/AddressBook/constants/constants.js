export const LISTBOX_PADDING = 8;
export const DOUBLE_LISTBOX_PADDING = LISTBOX_PADDING * 2;
export const LIMIT_NUMBER_ITEMS = 8;
export const ITEM_SIZE_LG = 48;
export const ITEM_SIZE_MD = 36;
