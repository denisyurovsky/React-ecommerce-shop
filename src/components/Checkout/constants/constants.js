export const PANEL = {
  DELIVERY_ADDRESS: 'Delivery Address',
  PERSONAL_INFORMATION: 'Personal Information',
  PERSONAL_ADDRESS: 'Personal Address',
  PAYMENT_METHOD: 'Payment method',
  ORDER_CONFIRMATION: 'Order confirmation',
};
