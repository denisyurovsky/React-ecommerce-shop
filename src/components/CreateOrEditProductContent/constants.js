export const PRICE_ERROR = 'Only numbers';
export const DESCRIPTION_ERROR = 'Only latin characters';
export const LENGTH_ERROR = 'Should contain at least one symbol';
export const DISCOUNT_ERROR = 'Discount should be less than price';
export const SELECTION_ERROR = 'Choose category';
