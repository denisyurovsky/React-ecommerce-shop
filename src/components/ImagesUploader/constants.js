export const LOSS_OF_IMAGES = 'We lost images';

export const STATUS = {
  CROPPED: 'cropped',
  CROPPING: 'cropping',
  IDLE: 'idle',
};

export const TEST_IMG_URL = 'http://someURL';
