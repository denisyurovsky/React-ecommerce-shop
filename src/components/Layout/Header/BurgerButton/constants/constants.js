import { pathNames } from '../../../../../helpers/constants/pathNames/pathNames';

export const LINKS = [
  {
    name: 'HOME',
    link: '/',
  },
  {
    name: 'PRODUCTS',
    link: pathNames.PRODUCTS,
  },
];
