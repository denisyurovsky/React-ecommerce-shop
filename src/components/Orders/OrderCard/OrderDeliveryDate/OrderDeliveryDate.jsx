import { Typography } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';

import {
  orderState,
  orderStatus,
} from '../../../../helpers/constants/orderStatus';
import { formatDateWithFullMonth } from '../../../../helpers/utils/formatData';

export const OrderDeliveryDate = ({
  status,
  deliveryDate,
  breakpointDisplayed = true,
}) => {
  return orderStatus[status].deliveryStatus ===
    orderStatus[orderState.DELIVERED].deliveryStatus && breakpointDisplayed ? (
    <Typography>
      Delivered on {formatDateWithFullMonth(deliveryDate)}
    </Typography>
  ) : (
    <></>
  );
};

OrderDeliveryDate.propTypes = {
  status: PropTypes.number,
  deliveryDate: PropTypes.string,
  breakpointDisplayed: PropTypes.bool,
};
