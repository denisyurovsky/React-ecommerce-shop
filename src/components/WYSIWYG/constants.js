export const TYPES = {
  LINK: 'LINK',
  CELL: 'CELL',
  EMAIL: 'EMAIL',
  TABLE: 'TABLE',
};

export const table = {
  MAX_ROWS: 20,
  COLUMNS: 2,
  HEADER: 'Product details',
  INPUT_ERROR: 'Max 20 rows',
};

export const MAX_LENGTH_MESSAGE = 'Max size is reached';
export const LINK_PREDEFINED_TEXT = 'https://';
export const EDITOR_MAX_SYMBOLS = 1000;
export const EDITOR_WARN_SYMBOLS = 50;

export const BREAKPOINTS = {
  FOOTER: 550,
  NAVIGATION: 350,
};
