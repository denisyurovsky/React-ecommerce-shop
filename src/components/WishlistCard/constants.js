export const DEFAULT_LABEL_INPUT = 'Wishlist Name';
export const ERROR_LABEL_INPUT = 'Wishlist name should be unique';
