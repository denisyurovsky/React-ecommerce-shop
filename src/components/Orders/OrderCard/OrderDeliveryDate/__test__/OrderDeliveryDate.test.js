import { render } from '@testing-library/react';
import React from 'react';

import { orderState } from '../../../../../helpers/constants/orderStatus';
import { OrderDeliveryDate } from '../OrderDeliveryDate';

describe('OrderDeliveryDate component', () => {
  describe('snapshots for different order statuses', () => {
    const now = new Date();

    it('Renders a valid orderCard snapshot with status equal WAITING_FOR_PAYMENT', () => {
      const { asFragment } = render(
        <OrderDeliveryDate
          status={orderState.WAITING_FOR_PAYMENT}
          deliveryDate={now}
        />
      );

      expect(asFragment()).toMatchSnapshot();
    });

    it('Renders a valid orderCard snapshot with status equal DELIVERED', () => {
      const { asFragment } = render(
        <OrderDeliveryDate status={orderState.DELIVERED} deliveryDate={now} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
    it('Renders a valid orderCard snapshot with status equal PAID', () => {
      const { asFragment } = render(
        <OrderDeliveryDate status={orderState.PAID} deliveryDate={now} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
    it('Renders a valid orderCard snapshot with status equal CANCELLED', () => {
      const { asFragment } = render(
        <OrderDeliveryDate status={orderState.CANCELLED} deliveryDate={now} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
