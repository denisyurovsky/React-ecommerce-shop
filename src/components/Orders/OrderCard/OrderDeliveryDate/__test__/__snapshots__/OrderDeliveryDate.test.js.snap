// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDeliveryDate component snapshots for different order statuses Renders a valid orderCard snapshot with status equal CANCELLED 1`] = `<DocumentFragment />`;

exports[`OrderDeliveryDate component snapshots for different order statuses Renders a valid orderCard snapshot with status equal DELIVERED 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Delivered on 19 April 2021
  </p>
</DocumentFragment>
`;

exports[`OrderDeliveryDate component snapshots for different order statuses Renders a valid orderCard snapshot with status equal PAID 1`] = `<DocumentFragment />`;

exports[`OrderDeliveryDate component snapshots for different order statuses Renders a valid orderCard snapshot with status equal WAITING_FOR_PAYMENT 1`] = `<DocumentFragment />`;
